power_grid_model_c/power_grid_model/include/power_grid_model/component/load_gen.hpp

```diff
@@ -60,11 +60,20 @@ class GenericGenerator : public GenericLoadGen {
 
 template <bool sym, bool is_gen>
 class LoadGen final : public std::conditional_t<is_gen, GenericGenerator, GenericLoad> {
+
+    static constexpr char const* get_name() {
+        if constexpr (sym) {
+            return is_gen ? "sym_gen" : "sym_load";
+        } else {
+            return is_gen ? "asym_gen" : "asym_load";
+        }
+    }
+
   public:
     using InputType = LoadGenInput<sym>;
     using UpdateType = LoadGenUpdate<sym>;
     using BaseClass = std::conditional_t<is_gen, GenericGenerator, GenericLoad>;
-    static constexpr char const* name = sym ? (is_gen ? "sym_gen" : "sym_load") : (is_gen ? "asym_gen" : "asym_load");
+    static constexpr char const* name = get_name();
```

Comment:
sonar cloud doesn't like this because it's a function that's not called at runtime (cause it's `constexpr`) and therefore "untested". Please change this to

```cpp
    static constexpr char const* name = []{
        if constexpr (sym) {
            return is_gen ? "sym_gen" : "sym_load";
        } else {
            return is_gen ? "asym_gen" : "asym_load";
        }
    }();
```